docs/transform.ipynb

```diff
@@ -116,61 +410,90 @@
     "id": "861ea1ada088"
    },
    "source": [
-    "Optimizers that come with cirq can be found in the `cirq.optimizers` package.\n",
+    "## Overview\n",
+    "Transformers that come with cirq can be found in the `cirq.transformers` package.\n",
     "\n",
     "A few notable examples are:\n",
-    "\n",
-    "*   **ConvertToCzAndSingleGates**: Attempts to convert a circuit into CZ gates and single qubit gates.  This uses gate's unitary and decompose methods to transform them into CZ + single qubit gates.\n",
-    "*   **DropEmptyMoments** / **DropNegligible**:  Removes moments that are empty or have very small effects, respectively.\n",
-    "*   **EjectPhasedPaulis**: Pushes X, Y, and PhasedX gates towards the end of the circuit, potentially absorbing Z gates and modifying gates along the way.\n",
-    "*   **EjectZ**:  Pushes Z gates towards the end of the circuit, potentially adjusting phases of gates that they pass through.\n",
-    "*   **ExpandComposite**:  Uses `cirq.decompose` to expand composite gates.\n",
-    "*   **MergeInteractions**:  Combines series of adjacent one and two-qubit gates acting on a pair of qubits.\n",
-    "*   **MergeSingleQubitGates**:  Combines series of adjacent unitary 1-qubit operations\n",
-    "*   **SynchronizeTerminalMeasurements**:  Moves all measurements in a circuit to the final moment if possible.\n"
+    "*   **cirq.align_left / cirq.align_right**: Align gates to the left/right of the circuit.\n",
+    "*   **cirq.defer_measurements**:  Moves all (non-terminal) measurements in a circuit to the end of circuit by implementing the deferred measurement principle.\n",
+    "*   **cirq.drop_empty_moments** / **cirq.drop_negligible_operations**:  Removes moments that are empty or operations that have very small effects, respectively.\n",
+    "*   **cirq.eject_phased_paulis**: Pushes X, Y, and PhasedX gates towards the end of the circuit, potentially absorbing Z gates and modifying gates along the way.\n",
+    "*   **cirq.eject_z**:  Pushes Z gates towards the end of the circuit, potentially adjusting phases of gates that they pass through.\n",
+    "*   **cirq.expand_composite**:  Uses `cirq.decompose` to expand composite gates.\n",
+    "*   **cirq.merge_k_qubit_unitaries**: Replaces connected components of unitary operations, acting on <= k qubits, with op-tree given by `rewriter(circuit_op)`.\n",
+    "*   **cirq.optimize_for_target_gateset**: Attempts to convert a circuit into and equivalent circuit using only gates from a given target gateset.\n",
+    "*   **cirq.stratified_circuit:**: Repacks the circuit to ensure that moments only contain operations from the same category.\n",
+    "*   **cirq.synchronize_terminal_measurements**:  Moves all terminal measurements in a circuit to the final moment, if possible."
```

Comment:
Personally I would delete the laundry list here and just link out to the `cirq.transformers` module so a motivated user could sift through the latest features there. Plus as maintainers we don't have to remember to keep this up to date.

Reply:
My motivation to keep this list as it is is that `cirq.transformers` reference page on the docs site has too many functions (eg: all functions from transformers.analytical_decompositions, transformers.heuristic_decompositions etc.) and it's hard to look at the reference page and guess which of these are transformers. 

Can we modify the reference page to have a classification of functions into transformers / transformer primitives / analytical decompositions etc?

Till we don't have that, I think it's worth keeping this list here.